commons/ui/src/main/kotlin/com/teacherworkout/commons/ui/extensions/NavigationExtension.kt

```diff
@@ -0,0 +1,242 @@
+package com.teacherworkout.commons.ui.extensions
+
+import android.content.Intent
+import android.util.SparseArray
+import androidx.core.util.forEach
+import androidx.core.util.set
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.navigation.NavController
+import androidx.navigation.fragment.NavHostFragment
+import com.google.android.material.bottomnavigation.BottomNavigationView
+
+/**
+ * This is a workaround until the Navigation Component supports multiple back stacks.
+ * Issue tracking: https://issuetracker.google.com/issues/127932815
+ * Issue github: https://github.com/android/architecture-components-samples/issues/530
+ */
+@Suppress("UnsafeCast", "UnsafeCallOnNullableType", "LongMethod", "ComplexMethod")
+fun BottomNavigationView.setupWithNavController(
+    navGraphIds: List<Int>,
+    fragmentManager: FragmentManager,
+    containerId: Int,
+    intent: Intent
+): LiveData<NavController> {
```

Comment:
Not happy with this workaround at all, I need to spend some time going through the reasoning in both of the issue and pr and figure out what to do.

Any feedback will be greatly appreciated.

Reply:
Isn't this issue with multiple backstack [solved](https://medium.com/androiddevelopers/multiple-back-stacks-b714d974f134) in newest alphas for navigation? This is why I used the newest alpha for the navigation dependency in my PR.